### ISSUE TITLE
GetDisplayName() and FullName use different notation for nested classes

### ISSUE THREAD

**Issue by jtschuster** [dotnet/linker#2441]
The GetDisplayName extension method uses "." as path delimiters for nested classes, but the FullName member uses "/".

MessageContainer.cs appends either the GetDisplayName string or FullPath to the warnings, and strings passed in usually use GetDisplayName. 
```C#
if (Origin?.Provider is MethodDefinition method)
	sb.Append (method.GetDisplayName ());
else if (Origin?.Provider is IMemberDefinition member)
	sb.Append (member.FullName);
else if (Origin?.Provider is AssemblyDefinition assembly)
	sb.Append (assembly.Name.Name);
else
	throw new NotSupportedException ();
```

This leads to different looking paths in the same warning:
```
Trim analysis warning IL2109: Mono.Linker.Tests.Cases.RequiresCapability.RequiresCapability/RequiresOnClass/DerivedWithoutRequires: Type 'Mono.Linker.Tests.Cases.RequiresCapability.RequiresCapability.RequiresOnClass.DerivedWithoutRequires' derives from 'Mono.Linker.Tests.Cases.RequiresCapability.RequiresCapability.RequiresOnClass.ClassWithRequires' which has 'RequiresUnreferencedCodeAttribute'.  Message for --ClassWithRequires--.
```

**Comment by vitek-karas** [comment]
Everything written to the diagnostic message should use `GetDisplayName`. This is a bit complicated but basically there are 4 competing formats how to represent any member:
* Cecil's format - that's what `FullName` returns - this is implemented by Cecil itself. We also use this format in the [linker XML's](https://github.com/dotnet/linker/blob/main/docs/data-formats.md) since that's what linker has been doing since forever - and Cecil supports parsing such format.
* Reflection format - this is what runtime returns from properties like [`Type.FullName`](https://docs.microsoft.com/en-us/dotnet/api/system.type.fullname?view=net-6.0#:~:text=FullName%20%7B%20get%3B%20%7D%20The%20fully%20qualified%20name,generic%20type%20definition%20but%20contains%20unresolved%20type%20parameters.). Michal is actually advertising for use to use this format in the linker. There's definitely merit to that - the reflection format is universal (not language dependent) and easy to lookup when debugging. The linker should not be too language specific - currently it's very pro-C# and cons-Everything else.
* C#/Roslyn format - this is what Roslyn/C# uses - this more or less matches C# syntax (with some caveats). Currently this is what the `GetDisplayName` tries to produce - the thinking was to produce the same text as the Roslyn analyzers.
* Documentation format - we have a copy of the parser (and writer) for this from Roslyn [here](https://github.com/dotnet/linker/blob/main/src/linker/Linker/DocumentationSignatureParser.cs) - updated to work on top of Cecil. This is the format uses in for example `DynamicDependency` attribute (I think).

Currently diagnostics is trying to use the C#/Roslyn format, so all messages should use `GetDisplayName` instead of `ToString` or `FullName` (which would produce Cecil's representation).

**Comment by jtschuster** [comment]
Closed by PR #2449
